lib/lf/uscalfe/loc_comp_ellbvp.h

```diff
@@ -693,17 +752,46 @@ class ScalarLoadEdgeVectorProvider {
    *
    * @param fe_edge_p FE specification on edge
    * @param g functor object providing edge data
-   * @param edge_sel selector predicate for active edges. 
+   * @param edge_sel selector predicate for active edges.
+   *
+   * This constructor selects one of LehrFEM++'s built-in quadrature rules
+   * with a degree of exactness twice as big as the polynomial degree of the
+   * finite element space.
    */
   ScalarLoadEdgeVectorProvider(
       std::shared_ptr<const UniformScalarFESpace<SCALAR>> fe_space, FUNCTOR g,
       EDGESELECTOR edge_sel = base::PredicateTrue{})
       : g_(std::move(g)), edge_sel_(std::move(edge_sel)), pfe_() {
     auto fe = fe_space->ShapeFunctionLayout(base::RefEl::kSegment());
-    LF_ASSERT_MSG(fe, "Reference Finite Element for segments not available");
+    LF_ASSERT_MSG(fe != nullptr, "No shape functions specified for edges");
+    // Precompute entity-independent quantities based on a LehrFEM++ built-in
+    // quadrature rule
     pfe_ = PrecomputedScalarReferenceFiniteElement(
         fe, quad::make_QuadRule(base::RefEl::kSegment(), 2 * fe->Degree()));
+    LF_ASSERT_MSG(pfe_.isInitialized(),"PFE object not initialized!");
```

Comment:
I think this line is not needed: we check above that fe != nullptr, hence pfe_.isInitialized() == false always.